internal/version/main.go

```diff
@@ -0,0 +1,51 @@
+package version
+
+import (
+	"fmt"
+	"runtime"
+)
+
+var (
+	version   string
+	buildDate string
+	tempo     string
```

Comment:
The operator has a config to set the tempo image:
https://github.com/os-observability/tempo-operator/blob/3f19e127d4f608de628c14cc0f48c433540f8208/config/manager/controller_manager_config.yaml#L22-L24

and in the status field we parse the tag of the image: https://github.com/os-observability/tempo-operator/blob/3f19e127d4f608de628c14cc0f48c433540f8208/controllers/tempo/microservices_controller.go#L216-L221

I think we should also set the tempo version here based on the image.

Reply:
Good suggestion! I'll implement it

Reply:
Or should we make the version configurable in the CRD or the operator configuration, and set the docker image based on this (new) version field?

I'm not sure how configurable the version should be, as the operator will always depend on a specific version. The images field is handy if people use their own containers (which need to be compatible with the version X we tested the operator with).

Reply:
> Or should we make the version configurable in the CRD or the operator configuration, and set the docker image based on this (new) version field (see @rubenvp8510's comment)?

If we decide to get the version from the CRD or the operator configuration, the scenario is "closed" since the user will not be able to provide their own images, right? If we set the container image based on the version field, we can allow to set directly the images as the OpenTelemetry Operator does, right?

I was not sure about what approach to follow... so I decided to follow the first one. I think is something we can decide later if we see it is a use case we should cover. What do you think?

Reply:
My understanding is that the operator in version x also only supports tempo un of version x. As well as the upgrade from x-1. As @andreasgerstmayr mentioned we can print the images from the spec.

Reply:
I changed the code behavior before the comment to print the images from the spec so, we're good to go.